src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -335,13 +335,23 @@ private static bool IsOSVersionAtLeast(int major, int minor, int build, int revi
             {
                 return current.Minor > minor;
             }
+            if (current.Build < 0)
+            {
+                // Unspecified component satisfies any required version
+                return true;
```

Comment:
```suggestion
                // Unspecified build component is to be treated as zero
                return build >= 0;
```

@filipnavara, like this?

Reply:
`build == 0` I think. (unless we change the version number normalisation which I would prefer in order to match iOS)

Reply:
> Suppose that Apple releases iOS 18.1.1, is macOS with version 18.1 behaving like 18.1.0 or does it behave like the patched version? I'd argue that it's safer to assume that it behaves like 18.1.0

This makes sense, but the following code will return `false`:

```csharp
Version current = Environment.OSVersion.Version;
OperatingSystem.IsOSPlatformVersionAtLeast(platformName, current.major, current.minor, current.build, current.revision)
```

I don't think it is expected behavior.